slm_porosity/heat_transfer.f90

```diff
@@ -826,4 +865,75 @@ FUNCTION Spline_cubic (r_p, l_p, fr_p, fl_p, dfr_p, dfl_p)
       - fr_p*l_p**3 - r_p**2*(3*fl_p*l_p - dfl_p*l_p**2 + dfr_p*l_p**2))/(r_p-l_p)**3
   END FUNCTION Spline_cubic
 
+  FUNCTION three_parameter_model (temperature, liquid_fraction, Dsolid, Dliquid, fusion_jump, is_d)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) ::  temperature(:), liquid_fraction(:)
+    REAL (pr) :: three_parameter_model(SIZE(temperature))
+    REAL (pr) :: Dsolid, Dliquid, fusion_jump
+    CHARACTER(LEN = 4), OPTIONAL, INTENT(IN) :: is_d
+
+    three_parameter_model = 1.0_pr + Dsolid*temperature + &
+      (fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr))*liquid_fraction
+    IF (is_d.EQ.'dphi') THEN
+      three_parameter_model = fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr)
+    ELSEIF (is_d.EQ.'dTem') THEN
+      three_parameter_model = Dsolid + (Dliquid - Dsolid)*liquid_fraction
+    END IF
+  END FUNCTION three_parameter_model
+
+  ! FUNCTION Dthree_parameter_model (temperature, liquid_fraction, Dtemperature, Dliquid_fraction, Dsolid, Dliquid, fusion_jump)
+  !   IMPLICIT NONE
+  !   REAL (pr), INTENT(IN) :: Dtemperature(:), Dliquid_fraction(:), liquid_fraction(:), temperature(:)
+  !   REAL (pr) :: Dthree_parameter_model(SIZE(temperature))
+  !   REAL (pr) :: Dsolid, Dliquid, fusion_jump
+  !
+  !   Dthree_parameter_model = (fusion_jump + (Dliquid - Dsolid)*(temperature - 1.0_pr))*Dliquid_fraction + &
+  !     (Dliquid - Dsolid)*Dtemperature*liquid_fraction + Dsolid*Dtemperature
+  ! END FUNCTION Dthree_parameter_model
+
+  FUNCTION Dirichlet_bc (enthalpy)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) :: enthalpy(:)
+    REAL (pr) :: Dirichlet_bc(SIZE(enthalpy))
+    REAL (pr) :: T(SIZE(enthalpy)), DT(SIZE(enthalpy))
+
+    T = temperature(enthalpy)
+    DT = Dtemperature(enthalpy, T)
+    Dirichlet_bc = (convective_transfer + emissivity*Stefan_Boltzmann*dim*T**dim)*DT
+  END FUNCTION Dirichlet_bc
+
+  FUNCTION heat_transfer_bc (enthalpy)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) :: enthalpy(:)
+    REAL (pr) :: heat_transfer_bc(SIZE(enthalpy))
+    REAL (pr) :: T(SIZE(enthalpy)), DT(SIZE(enthalpy))
+
+    T = temperature(enthalpy)
+    DT = Dtemperature(enthalpy, T)
+    heat_transfer_bc = - F_outer_heat(T) + F_outer_heat(T, .TRUE.)*DT*enthalpy
+  END FUNCTION heat_transfer_bc
+
+  FUNCTION F_outer_heat (temperature, is_D)
```

Comment:
We have a rule: do not use variables for global names. Here, just do not use `F`. If you hate `heat_transfer_bc` as I suggested many times before, you can call it `external_heat_flux`. `F` is not a heat, `F` is a heat flux.

Reply:
Didn't get it really. As we talked before the idea was to make two different function for `F` and for `heat_transfer_bc` for closer approach to metodichka.

Reply:
No, I meant that we need only function for F as in manual. Moreover, I said that we should have 3 terms in BC instead of 2 as in the manual.